src/glass/src/app/shared/services/local-inventory.service.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Project Aquarium's frontend (glass)
+ * Copyright (C) 2021 SUSE, LLC.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+
+import {
+  Inventory,
+  LocalNodeService,
+  Nic,
+  NodeStatus,
+  Volume
+} from './api/local.service';
+import { PollService } from './poll.service';
+
+export type NICEntry = {
+  name: string;
+  info: Nic;
+};
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LocalInventoryService {
+
+  private subject: BehaviorSubject<Inventory|undefined> =
+    new BehaviorSubject<Inventory|undefined>(undefined);
+
+  constructor(
+    private localService: LocalNodeService,
+    private pollService: PollService
+  ) {
+    this.localService
+      .status()
+      .pipe(
+        this.pollService.poll((status: NodeStatus) => !status.inited)
+      )
+      .subscribe({
+        next: (status: NodeStatus) => {
+          console.log('checking node status: ', status);
+          if (status.inited) {
+            this.obtainInventory();
+          }
+        }
+      });
+  }
+
+  getDevices(): Observable<Volume[]> {
+    return new Observable<Volume[]>( (observer) => {
```

Comment:
Please use `rxjs` to simplify the code.
```
return this.subject.pipe(
      map((inventory?: Inventory) => {
        if (!inventory) {
          return [];
        }
        return inventory.disks;
      })
    );
```
You need to add `import { map } from 'rxjs/operators';`, too.